src/matrix_graph.rs

```diff
@@ -853,16 +853,19 @@ fn extend_flat_square_matrix<T: Default>(
     for c in (1..old_node_capacity).rev() {
         let pos = c * old_node_capacity;
         let new_pos = c * new_node_capacity;
-
         // Move the slices directly if they do not overlap with their new position
         if pos + old_node_capacity <= new_pos {
-            let old = node_adjacencies[pos..pos + old_node_capacity].as_mut_ptr();
-            let new = node_adjacencies[new_pos..new_pos + old_node_capacity].as_mut_ptr();
```

Comment:
FYI my usual solution to this is
```rust
let ptr = node_adjacencies.as_mut_ptr();
let old = ptr.add(pos);
let new = ptr.add(new_pos);
```
This has zero overhead, and if anything is faster because it omits the bounds checks. Unclear if those are desired, but you could add them back in explicitly if you want.